raiden/tests/utils/protocol.py

```diff
@@ -75,8 +75,12 @@ def __init__(self, wrapped_handler: EventHandler):
         self.wrapped = wrapped_handler
         self.eventtype_to_waitingholds: Dict[type, List[HoldWait]] = defaultdict(list)
         self.eventtype_to_holdings: Dict[type, List[Holding]] = defaultdict(list)
+        self.pre_hooks: Set[Callable] = set()
```

Comment:
You don't need the hooks here because `on_raiden_event` will set the `AsyncResult` you get back when you want to `hold` a certain event. What you could do is `async_result.wait()` and once this happens you can do the asserts you're doing. Notice that you needed the hook to execute something before the raiden_event_handler dispatches the event forward. However, if you're paying attention to the code in `on_raiden_event`, the event won't be dispatched forward if "held" until you release it.